src/java/za/co/mip/ablduck/utilities/CommentParser.java

```diff
@@ -0,0 +1,239 @@
+        
+        //Take the internal comments out for javaDoc parser
+        protectInternalComments();
+        
+        //Generate links
+        generateLinks();
+        
+        //Deprecated flag
+        parseDeprecated(commentParseResult);
+        
+        //Internal tag
+        parseInternal(commentParseResult);
+        
+        //Return tag
+        parseReturn(commentParseResult);
+        
+        //Parameter tags
+        parseParamComments(commentParseResult);
+        
+        //Internal tag
+        parseAuthor(commentParseResult);
+        
+        //Put comments back in after javaDoc parser
+        returnInternalComments();
+        
+        comment = markdown(comment);
+        
+        commentParseResult.comment = comment;
+        
+        return commentParseResult;
+    }
+    
+    public void parseReturn(CommentParseResult commentParseResult){
+        commentParseResult.returnComment = getValueTag(Pattern.compile("@return\\s+(.*)", Pattern.DOTALL));
+    }
+    
+    public void parseAuthor(CommentParseResult commentParseResult){
+        commentParseResult.author = getValueTag(Pattern.compile("@author\\s+(.*)", Pattern.DOTALL));
+    }
+    
+    public void parseInternal(CommentParseResult commentParseResult){
+        commentParseResult.internal = getBooleanTag(Pattern.compile("@internal"));
+    }
+    
+    public void parseDeprecated(CommentParseResult commentParseResult) throws IOException{
+        HashMap<String, String> deprecated = getKeyValueTags(Pattern.compile("@deprecated\\s+(.+?)\\s+(.*)", Pattern.DOTALL));
+        
+        if (deprecated.size() > 0) {
+            commentParseResult.deprecatedVersion = deprecated.keySet().iterator().next();
+            commentParseResult.deprecatedText = deprecated.get(commentParseResult.deprecatedVersion);
+        }
+    }
+    
+    public void parseParamComments(CommentParseResult commentParseResult) throws IOException{
+        commentParseResult.parameterComments = getKeyValueTags(Pattern.compile("@param\\s+(.+?)\\s+(.*)", Pattern.DOTALL));
+    }
+    
+    public void protectInternalComments(){
+        Pattern simpleComment = Pattern.compile("(\\/\\*.*\\*\\/)", Pattern.DOTALL);
+        Matcher m = simpleComment.matcher(comment);
+        Integer commentCount = 0;
+        while (m.find()) {
+            String commentTag = "[comment-" + commentCount.toString() + "]";
+            nestedComments.put(commentTag, m.group(1));
+            comment = comment.replaceFirst(Pattern.quote(m.group(1)), Matcher.quoteReplacement(commentTag));
+            commentCount++;
+        }
+    }
+    
+    public void returnInternalComments(){
+        for (Map.Entry<String, String> c:nestedComments.entrySet()) {
+            comment = comment.replaceFirst(Pattern.quote(c.getKey()), Matcher.quoteReplacement(c.getValue()));
+        }
+    }
+    
+    public HashMap<String, String> getKeyValueTags(Pattern paramRegex) throws IOException{

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1319)